### PR TITLE
fix: do not override labels when not configured

### PR DESCRIPTION
**Pull request by viceice**
## Changes:

Do not override issue labels if labels are not configured (`undefined` or `null`). To force empty labels use an empty array.

Regression of #10601

## Context:

see #10798

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — APPROVED** [review]

**Review by bobvandevijver — APPROVED** [review]

File: lib/platform/github/index.ts [hunk 1]
```diff
@@ -1202,12 +1202,16 @@ export async function ensureIssue({
       }
       if (shouldReOpen) {
         logger.debug('Patching issue');
+        const data: Record<string, unknown> = { body, state: 'open', title };
+        if (labels) {
```
Comment (rarkins) [inline lib/platform/github/index.ts:1206]: @viceice Should be `if (labels?.length) {` ?

Comment (viceice) [reply lib/platform/github/index.ts:1206]: not sure, is `labels`  always defined? i thought it's undefined by default 🤔

Comment (rarkins) [reply lib/platform/github/index.ts:1206]: Something's not working. Can we go with a more conservative approach?

Comment (viceice) [reply lib/platform/github/index.ts:1206]: IÄll try to run some debug sessions, to see why it's again removing the labels when it shouldn't. Maybe we've a bug somewhere.

Comment (rarkins) [reply lib/platform/github/index.ts:1206]: My guess is that the platform layer is receiving an empty array

**Pull request merged**